FRW/Modules/Login/KeyStoreLoginView.swift

```diff
@@ -26,13 +26,22 @@ struct KeyStoreLoginView: RouteableView {
                         .padding(.top, 48)
 
                     Section {
-                        ImportTextView(
-                            content: $viewModel.json,
-                            placeholder: "keystore_json".localized
-                        ) { value in
-                            viewModel.update(json: value)
+                        VStack(spacing: 0) {
+                            ImportTextView(
+                                content: $viewModel.json,
+                                placeholder: "keystore_json".localized
+                            ) { value in
+                                viewModel.update(json: value)
+                            }
+                            .frame(height: 120)
+
+                            CalloutView(
+                                corners: [.bottomLeading, .bottomTrailing],
+                                content: viewModel.errorJSON
+                            )
+                            .padding(.horizontal, 12)
+                            .visibility(viewModel.errorJSON.isEmpty ? .gone : .visible)
```

Comment:
The logic to determine whether to display or not does not belong to the UI.

 The view model should expose a `hasJsonError` computed property, and the logical expression `errorJSON.isEmpty` should be moved there. 

This is a very simple case, but the logical expression could be more complex, and cases like this should be handled by the view model.

Reply:
removed code

Reply:
Have you moved this code elsewhere or removed it completely?